config/config.go

```diff
@@ -171,41 +171,41 @@ func (l *ListenConfig) UnmarshalText(data []byte) error {
 	return nil
 }
 
-// UnmarshalYAML creates BootstrapDNSConfig from YAML
-func (b *BootstrapDNSConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	var single BootstrappedUpstreamConfig
+// UnmarshalYAML creates BootstrapDNS from YAML
+func (b *BootstrapDNS) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var single BootstrappedUpstream
 	if err := unmarshal(&single); err == nil {
-		*b = BootstrapDNSConfig{single}
+		*b = BootstrapDNS{single}
 
 		return nil
 	}
 
 	// bootstrapDNSConfig is used to avoid infinite recursion:
-	// if we used BootstrapDNSConfig, unmarshal would just call us again.
-	var c bootstrapDNSConfig
+	// if we used BootstrapDNS, unmarshal would just call us again.
+	var c bootstrapDNS
 	if err := unmarshal(&c); err != nil {
 		return err
 	}
 
-	*b = BootstrapDNSConfig(c)
+	*b = BootstrapDNS(c)
 
 	return nil
 }
 
 // UnmarshalYAML creates BootstrapConfig from YAML
-func (b *BootstrappedUpstreamConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+func (b *BootstrappedUpstream) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := unmarshal(&b.Upstream); err == nil {
 		return nil
 	}
 
 	// bootstrapConfig is used to avoid infinite recursion:
 	// if we used BootstrapConfig, unmarshal would just call us again.
```

Comment:
```suggestion
	// bootstrappedUpstream is used to avoid infinite recursion:
	// if we used BootstrappedUpstream, unmarshal would just call us again.
```